Adds toy example and solves conflicts in README.md (#243)
rmcolq/pandora: toy_example/README.md

# Toy example

> Here we present a walkthrough of running `pandora` on a toy example. We run: 1) `pandora` without de novo discovery; 2) `pandora` workflow (https://github.com/iqbal-lab-org/pandora_workflow), which runs pandora with and without de novo discovery (see Figure 2 of Colquhoun et al. "Nucleotide-resolution bacterial pan-genomics with reference graphs"). Although method 2) run both modes of `pandora`, it is much more involved than method 1), as the user needs to configure and run a `snakemake` pipeline. If no de novo discovery is required, method 1 is a lot simpler to run (two command lines) as opposed to running the pipeline. For completeness, we show both methods.

Reviewer: ```suggestion
Here we present a walkthrough of running `pandora` on a toy example. We run: 1) `pandora` without de novo discovery; 2) [`pandora` workflow](https://github.com/iqbal-lab-org/pandora_workflow), which runs `pandora` with and without de novo discovery (see Figure 2 of Colquhoun et al. "Nucleotide-resolution bacterial pan-genomics with reference graphs"). Although method 2) runs both modes of `pandora`, it is much more involved than method 1), as the user needs to configure and run a `snakemake` pipeline. If no de novo discovery is required, method 1 is a lot simpler to run (two commands) as opposed to running the pipeline. For completeness, we show both methods.
```

Reviewer: I would either link to the preprint when we refer to it in the text, or just say "our paper"

Author: Will add link

## Input data description

```
msas/ : contains the MSAs of the two genes we are using as toy example here, GC00006032 and GC00010897;
prgs/toy_prg.fa : contains the PRGs of the two genes, GC00006032 and GC00010897. These are fairly simple PRGs. GC00006032 contains 4 variant sites, each representing a SNP, while GC00010897 contains 5;
reads/ : contains perfect simulated reads from two toy samples. We simulated perfect reads, where one sample genotypes to one allele of the variant sites, while the other sample genotypes towards the other allele;

> pandora_workflow_data : contains other input and configuration files to run the pandora workflow;

Reviewer: ```suggestion
pandora_workflow_data/ : contains other input and configuration files to run the pandora workflow;
```

```

##  pandora without de novo discovery

### Dependencies

> * `singularity`

Reviewer: ```suggestion
* [`singularity`](https://sylabs.io/)
```

### Running

> `cd toy_example && ./run_pandora_nodenovo.sh`

Reviewer: Could you put this in a shell code fence? I tried doing it in the suggestions but it screws it up due to the code fence escaping the suggestion

### Quick look at the output

> `pandora` output will be located at dir `output_toy_example_no_denovo`.

Reviewer: ```suggestion
`pandora` output will be located in directory `output_toy_example_no_denovo`.
```

GC00006032	160	.	A	C	.	.	SVTYPE=SNP;GRAPHTYPE=SIMPLE	GT:MEAN_FWD_COVG:MEAN_REV_COVG:MED_FWD_COVG:MED_REV_COVG:SUM_FWD_COVG:SUM_REV_COVG:GAPS:LIKELIHOOD:GT_CONF	1:2,3:3,9:0,5:0,11:6,19:11,47:0.666667,0.2:-61.987,-28.0629:33.9241	0:11,0:10,0:14,0:11,0:35,0:30,0:0.333333,1:-28.0849,-192.709:164.624
GC00006032	218	.	T	C	.	.	SVTYPE=SNP;GRAPHTYPE=SIMPLE	GT:MEAN_FWD_COVG:MEAN_REV_COVG:MED_FWD_COVG:MED_REV_COVG:SUM_FWD_COVG:SUM_REV_COVG:GAPS:LIKELIHOOD:GT_CONF	1:0,0:1,4:0,0:0,4:0,0:4,8:0.75,0:-28.725,-7.0005:21.7245	0:8,0:6,0:9,0:8,0:33,0:27,0:0.25,1:-30.9944,-160.472:129.478
```

We can see samples `toy_sample_1` and `toy_sample_2` genotype towards different alleles.

##  pandora workflow

### Dependencies

* `singularity`
* `git`
* `python 3.6+`

### Running
`cd toy_example && ./run_pandora_workflow.sh`

### Quick look at the output

`pandora` workflow output will be located at dir `pandora_workflow/output_toy_example_workflow/illumina/100x/random/compare_(no|with)denovo_global_genotyping`.

> Files `pandora_workflow/output_toy_example_workflow/illumina/100x/random/compare_nodenovo_global_genotyping/pandora_multisample_genotyped_global.vcf` and `output_toy_example_no_denovo/pandora_multisample_genotyped.vcf` both represent running `pandora` without de novo discovery and should be very similar files (just differentiating on some header lines, and on some statistics, due to slightly different versions used. For the first file, it is the version used in the paper; for the second file, the version on the `master` branch, which is more recent, with some bugs fixed). 

Reviewer: The super long parentheses don't need to be in parentheses...

```
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	toy_sample_1	toy_sample_2
GC00006032	146	.	C	T	.	.	SVTYPE=SNP;GRAPHTYPE=SIMPLE	GT:MEAN_FWD_COVG:MEAN_REV_COVG:MED_FWD_COVG:MED_REV_COVG:SUM_FWD_COVG:SUM_REV_COVG:GAPS:LIKELIHOOD:GT_CONF	0:41,0:52,0:41,0:52,0:83,0:105,0:0,1:-18.7786,-526.281:507.502	1:0,15:0,15:0,15:0,15:0,31:0,31:1,0:-214.155,-3.53065:210.624
GC00006032	160	.	A	C	.	.	SVTYPE=SNP;GRAPHTYPE=SIMPLE	GT:MEAN_FWD_COVG:MEAN_REV_COVG:MED_FWD_COVG:MED_REV_COVG:SUM_FWD_COVG:SUM_REV_COVG:GAPS:LIKELIHOOD:GT_CONF	1:14,29:17,42:0,36:0,51:42,148:52,212:0.666667,0.2:-366.08,-159.943:206.137	0:12,0:8,0:18,0:9,0:38,0:24,0:0.333333,1:-20.2506,-168.103:147.853
GC00006032	218	.	T	C	.	.	SVTYPE=SNP;GRAPHTYPE=SIMPLE	GT:MEAN_FWD_COVG:MEAN_REV_COVG:MED_FWD_COVG:MED_REV_COVG:SUM_FWD_COVG:SUM_REV_COVG:GAPS:LIKELIHOOD:GT_CONF	1:3,11:4,14:0,11:0,14:12,23:16,28:0.75,0:-182.162,-41.9443:140.217	0:11,0:5,0:13,0:6,0:44,0:21,0:0.25,1:-19.9705,-149.683:129.712
```

We can see samples `toy_sample_1` and `toy_sample_2` genotype towards different alleles.

##  pandora workflow

### Dependencies

* [`singularity`](https://sylabs.io/)
* `git`
* `python 3.6+`

### Running
```

> cd toy_example && ./run_pandora_workflow.sh

Reviewer: This works now
